libbpf-tools/llcstat.c

```diff
@@ -185,24 +188,23 @@ int main(int argc, char **argv)
 		return 1;
 	}
 
-	obj = llcstat_bpf__open();
-	if (!obj) {
-		fprintf(stderr, "failed to open BPF object\n");
+	nr_cpus = libbpf_num_possible_cpus();
+	if (nr_cpus < 0) {
+		fprintf(stderr, "failed to get # of possible cpus: '%s'!\n",
+			strerror(-nr_cpus));
 		return 1;
 	}
-
-	nr_cpus = libbpf_num_possible_cpus();
 	mlinks = calloc(nr_cpus, sizeof(*mlinks));
 	rlinks = calloc(nr_cpus, sizeof(*rlinks));
 	if (!mlinks || !rlinks) {
 		fprintf(stderr, "failed to alloc mlinks or rlinks\n");
-		goto cleanup;
+		return 1;
 	}
 
-	err = llcstat_bpf__load(obj);
-	if (err) {
-		fprintf(stderr, "failed to load BPF object: %d\n", err);
-		goto cleanup;
+	obj = llcstat_bpf__open_and_load();
+	if (!obj) {
+		fprintf(stderr, "failed to open and/or load BPF object\n");
+		return 1;
```

Comment:
why not `goto cleanup;` here? free(mlinks) and free(rlinks) would be good

Reply:
Yes, it should be `goto cleanup`, fixed, thanks